core/src/test/java/com/yahoo/imapnio/async/data/SearchResultTest.java

```diff
@@ -16,21 +16,35 @@
      */
     @Test
     public void testSearchResult() {
-        final List<Long> ll = new ArrayList<>();
-        ll.add(Long.MAX_VALUE - 1);
-        final SearchResult infos = new SearchResult(ll);
+        final List<Long> ll = Collections.singletonList(Long.MAX_VALUE - 1);
+        final SearchResult infos = new SearchResult(ll, 1L);
         final List<Long> result = infos.getMessageNumbers();
+        final long modSeq = infos.getHighestModSeq();
         Assert.assertEquals(result.size(), 1, "Result mismatched.");
         Assert.assertEquals(result.get(0), Long.valueOf(Long.MAX_VALUE - 1), "Result mismatched.");
+        Assert.assertEquals(modSeq, 1L, "Result mismatched.");
+    }
+
+    /**
+     * Tests SearchResult constructor and getters when passing null list and not passing modification sequence.
+     */
+    @Test
+    public void testSearchResultNoFiled() {
+        final SearchResult infos = new SearchResult();
+        Assert.assertNull(infos.getMessageNumbers(), "Result mismatched.");
```

Comment:
Again, Using "Result mismatched" for all asserts is not advisable, please use appropriate messages in all tests.